pkg/kubelet/cm/deviceplugin/endpoint_store_stub.go

```diff
@@ -0,0 +1,64 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package deviceplugin
+
+type endpointStoreStub struct {
+	endpointStore store
+
+	outChan      chan interface{}
+	continueChan chan bool
+}
+
+type replaceMessage struct {
+	Old *endpoint
+	New *endpoint
+}
+
+// ChanOut sends a message containing the parameters of the method called.
+// After sending that message, execution blocks until ChanContinue is written on
+func newInstrumentedEndpointStoreStub(outChan chan interface{}, continueChan chan bool) store {
```

Comment:
I think I get it - so this is in the absence of a mocked interface to the endpoints?
First question would be if this is the idiomatic way of getting this done. Let's take a look around and see if this hasn't been solved before.

Reply:
I think you raise a good point about looking at how this is done :)

The idea behind this was that I wanted to check if at the moment an endpoint is persisted in storage, we advertise devices or not.
Meaning that if a device plugin re-registers we don't remove the number of devices it advertises during re-registration.